### PR TITLE
Update libcontainer

### LINKED ISSUE #12130: Container port not exposed on the host with 1.6 rc4
A docker config that works on 1.5 fails with the latest 1.6 rc. The container that exposed the problem is running etcd built from scratch. Its exposes ports '2380' and '4001' on the host
```shell
        "NetworkMode": "container:4cff5d962ae7fb553fd7af413e06150bb9d0d9e527ae7951977b7d591e0cc42f",
        "PidMode": "",
        "PortBindings": {
            "2380/tcp": [
                {   
                    "HostIp": "",
                    "HostPort": "2380"
                }
            ],
            "4001/tcp": [
                {   
                    "HostIp": "",
                    "HostPort": "4001"
                }
            ]
        },
        "Privileged": false,
        "PublishAllPorts": false,
```

The port does not show up on the host network
```shell
netstat -ntlp
(No info could be read for "-p": geteuid()=1002 but you should be root.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      -               
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      -               
tcp        0      0 127.0.0.1:10248         0.0.0.0:*               LISTEN      -               
tcp        0      0 10.240.241.220:7080     0.0.0.0:*               LISTEN      -               
tcp        0      0 127.0.0.1:10251         0.0.0.0:*               LISTEN      -               
tcp        0      0 10.240.241.220:6443     0.0.0.0:*               LISTEN      -               
tcp        0      0 127.0.0.1:10252         0.0.0.0:*               LISTEN      -               
tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN      -               
tcp6       0      0 :::22                   :::*                    LISTEN      -               
tcp6       0      0 :::4194                 :::*                    LISTEN      -               
tcp6       0      0 :::10250                :::*                    LISTEN      -               
tcp6       0      0 :::10255                :::*                    LISTEN      -   
```

Once I switch to 1.5 the ports show up on the host.

Complete inspect output:
```shell
$ sudo docker inspect 0ade0593b217
[{
    "AppArmorProfile": "",
    "Args": [
        "--addr",
        "127.0.0.1:4001",
        "--bind-addr",
        "127.0.0.1:4001",
        "--data-dir",
        "/var/etcd/data"
    ],
    "Config": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": null,
        "CpuShares": 0,
        "Cpuset": "",
        "Domainname": "",
        "Entrypoint": [
            "/usr/local/bin/etcd",
            "--addr",
            "127.0.0.1:4001",
            "--bind-addr",
            "127.0.0.1:4001",
            "--data-dir",
            "/var/etcd/data"
        ],
        "Env": [
            "KUBE_DNS_SERVICE_HOST=10.0.0.10",
            "MONITORING_HEAPSTER_SERVICE_HOST=10.0.111.163",
            "KUBERNETES_RO_SERVICE_HOST=10.0.0.1",
            "KUBERNETES_RO_PORT_80_TCP=tcp://10.0.0.1:80",
            "KUBERNETES_PORT_443_TCP_PROTO=tcp",
            "MONITORING_GRAFANA_PORT_80_TCP_PROTO=tcp",
            "KIBANA_LOGGING_PORT_5601_TCP=tcp://10.0.55.106:5601",
            "MONITORING_HEAPSTER_PORT=tcp://10.0.111.163:80",
            "KUBERNETES_RO_PORT=tcp://10.0.0.1:80",
            "MONITORING_INFLUXDB_UI_SERVICE_PORT=80",
            "MONITORING_INFLUXDB_PORT_80_TCP_PROTO=tcp",
            "KUBE_DNS_PORT_53_UDP=udp://10.0.0.10:53",
            "MONITORING_HEAPSTER_PORT_80_TCP_ADDR=10.0.111.163",
            "MONITORING_GRAFANA_PORT_80_TCP_ADDR=10.0.110.242",
            "ELASTICSEARCH_LOGGING_PORT_9200_TCP_PROTO=tcp",
            "MONITORING_HEAPSTER_PORT_80_TCP_PORT=80",
            "KIBANA_LOGGING_PORT_5601_TCP_ADDR=10.0.55.106",
            "KUBE_DNS_SERVICE_PORT=53",
            "ELASTICSEARCH_LOGGING_SERVICE_HOST=10.0.94.161",
            "ELASTICSEARCH_LOGGING_SERVICE_PORT=9200",
            "KIBANA_LOGGING_PORT_5601_TCP_PROTO=tcp",
            "KUBERNETES_SERVICE_PORT=443",
            "KUBERNETES_PORT_443_TCP_PORT=443",
            "KUBERNETES_RO_PORT_80_TCP_ADDR=10.0.0.1",
            "KUBE_DNS_PORT_53_UDP_PROTO=udp",
            "MONITORING_INFLUXDB_UI_PORT_80_TCP_PROTO=tcp",
            "MONITORING_INFLUXDB_SERVICE_PORT=80",
            "KIBANA_LOGGING_PORT_5601_TCP_PORT=5601",
            "KUBE_DNS_PORT_53_UDP_ADDR=10.0.0.10",
            "MONITORING_HEAPSTER_PORT_80_TCP_PROTO=tcp",
            "MONITORING_GRAFANA_SERVICE_HOST=10.0.110.242",
            "MONITORING_INFLUXDB_UI_PORT_80_TCP=tcp://10.0.144.179:80",
            "KIBANA_LOGGING_PORT=tcp://10.0.55.106:5601",
            "MONITORING_GRAFANA_PORT_80_TCP_PORT=80",
            "KIBANA_LOGGING_SERVICE_PORT=5601",
            "KUBERNETES_PORT_443_TCP=tcp://10.0.0.2:443",
            "KUBERNETES_PORT_443_TCP_ADDR=10.0.0.2",
            "MONITORING_GRAFANA_PORT=tcp://10.0.110.242:80",
            "MONITORING_INFLUXDB_PORT_80_TCP=tcp://10.0.235.26:80",
            "KUBE_DNS_PORT_53_UDP_PORT=53",
            "KUBERNETES_RO_SERVICE_PORT=80",
            "MONITORING_INFLUXDB_UI_SERVICE_HOST=10.0.144.179",
            "MONITORING_INFLUXDB_PORT_80_TCP_PORT=80",
            "ELASTICSEARCH_LOGGING_PORT_9200_TCP_ADDR=10.0.94.161",
            "MONITORING_GRAFANA_PORT_80_TCP=tcp://10.0.110.242:80",
            "MONITORING_INFLUXDB_SERVICE_HOST=10.0.235.26",
            "KUBE_DNS_PORT=udp://10.0.0.10:53",
            "MONITORING_GRAFANA_SERVICE_PORT=80",
            "KIBANA_LOGGING_SERVICE_HOST=10.0.55.106",
            "MONITORING_HEAPSTER_PORT_80_TCP=tcp://10.0.111.163:80",
            "ELASTICSEARCH_LOGGING_PORT_9200_TCP=tcp://10.0.94.161:9200",
            "KUBERNETES_PORT=tcp://10.0.0.2:443",
            "MONITORING_INFLUXDB_UI_PORT_80_TCP_PORT=80",
            "ELASTICSEARCH_LOGGING_PORT=tcp://10.0.94.161:9200",
            "KUBERNETES_RO_PORT_80_TCP_PROTO=tcp",
            "KUBERNETES_SERVICE_HOST=10.0.0.2",
            "MONITORING_INFLUXDB_PORT=tcp://10.0.235.26:80",
            "ELASTICSEARCH_LOGGING_PORT_9200_TCP_PORT=9200",
            "MONITORING_HEAPSTER_SERVICE_PORT=80",
            "KUBERNETES_RO_PORT_80_TCP_PORT=80",
            "MONITORING_INFLUXDB_UI_PORT=tcp://10.0.144.179:80",
            "MONITORING_INFLUXDB_UI_PORT_80_TCP_ADDR=10.0.144.179",
            "MONITORING_INFLUXDB_PORT_80_TCP_ADDR=10.0.235.26",
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
        ],
        "ExposedPorts": {
            "2380/tcp": {},
            "4001/tcp": {}
        },
        "Hostname": "e2e-test-vishnuk-master",
        "Image": "gcr.io/google_containers/etcd:2.0.8",
        "Labels": {},
        "MacAddress": "",
        "Memory": 0,
        "MemorySwap": 0,
        "NetworkDisabled": false,
        "OnBuild": null,
        "OpenStdin": false,
        "PortSpecs": null,
        "StdinOnce": false,
        "Tty": false,
        "User": "",
        "Volumes": null,
        "WorkingDir": ""
    },
    "Created": "2015-04-06T23:46:46.61424277Z",
    "Driver": "aufs",
    "ExecDriver": "native-0.2",
    "ExecIDs": [
        "ff915b3d25aaa0cf9175d4e5368a7dfce44a641512c3557d4e55c457d503d937",
        "1b2b714d58966cf723126c6ef8abc4864becd05f699e62cece4bc43018a58656",
        "e024386fea75476daa99b8f002592bee1063df35a0edcd2871309de310984618"
    ],
    "HostConfig": {
        "Binds": [
            "/mnt/master-pd/var/etcd:/var/etcd",
            "/var/lib/kubelet/pods/160ec72a0805d5306a34deee7743e5b9/containers/etcd-container/0ade0593b217bffefe38b753c49d1b07f6efdaefc4686bf72d299833864b27e9:/dev/termination-log"
        ],
        "CapAdd": null,
        "CapDrop": null,
        "CgroupParent": "",
        "ContainerIDFile": "",
        "CpuShares": 0,
        "CpusetCpus": "",
        "Devices": null,
        "Dns": [
            "10.0.0.10",
            "169.254.169.254",
            "10.240.0.1"
        ],
        "DnsSearch": [
            "default.kubernetes.local",
            "kubernetes.local",
            "c.vishnuk-cloud.internal.",
            "454568409378.google.internal.",
            "google.internal."
        ],
        "ExtraHosts": null,
        "IpcMode": "container:4cff5d962ae7fb553fd7af413e06150bb9d0d9e527ae7951977b7d591e0cc42f",
        "Links": null,
        "LogConfig": {
            "Config": null,
            "Type": "json-file"
        },
        "LxcConf": null,
        "Memory": 0,
        "MemorySwap": 0,
        "NetworkMode": "container:4cff5d962ae7fb553fd7af413e06150bb9d0d9e527ae7951977b7d591e0cc42f",
        "PidMode": "",
        "PortBindings": {
            "2380/tcp": [
                {
                    "HostIp": "",
                    "HostPort": "2380"
                }
            ],
            "4001/tcp": [
                {
                    "HostIp": "",
                    "HostPort": "4001"
                }
            ]
        },
        "Privileged": false,
        "PublishAllPorts": false,
        "ReadonlyRootfs": false,
        "RestartPolicy": {
            "MaximumRetryCount": 0,
            "Name": ""
        },
        "SecurityOpt": null,
        "Ulimits": null,
        "VolumesFrom": null
    },
    "HostnamePath": "/var/lib/docker/containers/4cff5d962ae7fb553fd7af413e06150bb9d0d9e527ae7951977b7d591e0cc42f/hostname",
    "HostsPath": "/var/lib/docker/containers/4cff5d962ae7fb553fd7af413e06150bb9d0d9e527ae7951977b7d591e0cc42f/hosts",
    "Id": "0ade0593b217bffefe38b753c49d1b07f6efdaefc4686bf72d299833864b27e9",
    "Image": "7459fa04a08ca061a1ff8992a60d9364de2556aa2dd61ef2ee8c03500bddae0f",
    "LogPath": "",
    "MountLabel": "",
    "Name": "/k8s_etcd-container.1ab37454_etcd-server-e2e-test-vishnuk-master_default_160ec72a0805d5306a34deee7743e5b9_9838c56b",
    "NetworkSettings": {
        "Bridge": "",
        "Gateway": "",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "IPAddress": "",
        "IPPrefixLen": 0,
        "IPv6Gateway": "",
        "LinkLocalIPv6Address": "",
        "LinkLocalIPv6PrefixLen": 0,
        "MacAddress": "",
        "PortMapping": null,
        "Ports": null
    },
    "Path": "/usr/local/bin/etcd",
    "ProcessLabel": "",
    "ResolvConfPath": "/var/lib/docker/containers/4cff5d962ae7fb553fd7af413e06150bb9d0d9e527ae7951977b7d591e0cc42f/resolv.conf",
    "RestartCount": 0,
    "State": {
        "Dead": false,
        "Error": "",
        "ExitCode": 0,
        "FinishedAt": "0001-01-01T00:00:00Z",
        "OOMKilled": false,
        "Paused": false,
        "Pid": 19987,
        "Restarting": false,
        "Running": true,
        "StartedAt": "2015-04-06T23:46:46.761009987Z"
    },
    "Volumes": {
        "/dev/termination-log": "/var/lib/kubelet/pods/160ec72a0805d5306a34deee7743e5b9/containers/etcd-container/0ade0593b217bffefe38b753c49d1b07f6efdaefc4686bf72d299833864b27e9",
        "/var/etcd": "/mnt/master-pd/var/etcd"
    },
    "VolumesRW": {
        "/dev/termination-log": true,
        "/var/etcd": true
    }
}
]
```

cc @tiborvass @jfrazelle

### PR DESCRIPTION
**Pull request by LK4D4**
Fix #12130 
ping @vishh for test

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

File: integration-cli/docker_cli_run_test.go [hunk 1]
```diff
@@ -3249,6 +3249,35 @@ func TestRunNetHost(t *testing.T) {
 	logDone("run - net host mode")
 }
 
+func TestRunNetContainerWhichHost(t *testing.T) {
```
Comment (rjnagal) [inline integration-cli/docker_cli_run_test.go]: nit: s/WhichHost/WithHost/

**Comment by rjnagal** [comment]
LGTM
Thanks for the quick path, @LK4D4

**Comment by vishh** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**